cmake/dependencies.cmake

```diff
@@ -108,6 +108,18 @@ FetchContent_Declare(
   SYSTEM
   SOURCE_SUBDIR crates/c-api)
 
+FetchContent_GetProperties(ada)
+if(NOT ada_POPULATED)
+  # Disable git to prevent CPM from downloading 3rdparty repositories.
+  # An intentional side effect is that tests and tools are not built
+
+  FetchContent_Populate(ada)
+  set(GIT_EXECUTABLE_OLD ${GIT_EXECUTABLE})
+  set(GIT_EXECUTABLE "" CACHE INTERNAL "Invalid path to the Git executable")
+  add_subdirectory(${ada_SOURCE_DIR} ${ada_BINARY_DIR})
+  set(GIT_EXECUTABLE ${GIT_EXECUTABLE_OLD} CACHE STRING "location of git" FORCE)
```

Comment:
the simplest thing to do is fork ada and write a better cmake file. i did it for several of the dependencies in here, and it is pretty easy since we don't have maintain tests, benchmarks, installation, etc...

Reply:
alternatively, you can check-in a patch to the cmakefile and have fetchcontent apply the patch.

Reply:
I think @travisdowns knows one of the maintainers, so maybe we can get them to apply the patch (or just write a dumb CMake file for the single file version).

Reply:
yeh if we want to go through upstream that works. the problems with the ada cmake file are pretty reasonable i would expect maintainers to pick fixes for them. here is the one i did for Avro https://github.com/redpanda-data/avro/blob/release-1.11.1-redpanda/redpanda_build/CMakeLists.txt as an example of a custom cmakefile.

Reply:
@dotnwat  Any objection to merging this as-is to mitigate [this type of failure](https://buildkite.com/redpanda/redpanda/builds/40273#018b8dea-65c0-4a31-8c45-e0b3d41a1c30https://buildkite.com/redpanda/redpanda/builds/40273#018b8dea-65c0-4a31-8c45-e0b3d41a1c30) and then reworking the build in followup work?